packages/iov-core/src/jsrpcsigningserver.spec.ts

```diff
@@ -155,24 +156,23 @@ describe("JsonRpcSigningServer", () => {
   });
 
   it("can get BNS or Ethereum identities", async () => {
+    pendingWithoutBnsd();
     pendingWithoutEthereum();
 
     const bnsConnection = await bnsConnector(bnsdUrl).client();
 
-    const server = await makeJsonRpcSigningServer();
+    const server = await makeBnsEthereumSigningServer();
 
     const response = await server.handleChecked({
-      jsonrpc: "2.0",
       id: 123,
       method: "getIdentities",
       params: {
         reason: "Who are you?",
         chainIds: [ethereumChainId, bnsConnection.chainId()],
       },
     });
-    expect(response.jsonrpc).toEqual("2.0");
     expect(response.id).toEqual(123);
-    if (isJsonRpcErrorResponse(response)) {
+    if (isJsRpcErrorResponse(response)) {
```

Comment:
Ahh... I was confused why this was not using the client interface, but this is calling the server directly.

Maybe extract these lines to a function, same logic as in JsRpcClient.run.

Or even just use that client for these tests, as it seems a very light convenience wrapper around the server.
`await client.run(...)` instead of `await server.handleChecked(..)`. But this may break the test isolation, so feel free to leave as is... code to convert responses to `new Error` should be in one place though

Reply:
For the server, `JsRpcResponse = JsRpcSuccessResponse | JsRpcErrorResponse` are valid, non-error responses as described in https://www.jsonrpc.org/specification#response_object

It is the client's job to interprete both possible responses. In this test I expect a `JsRpcSuccessResponse` but in another test I might expect a `JsRpcErrorResponse`.